packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -1186,8 +1220,13 @@ export class AngularCompilerPlugin {
       ...this.getResourceDependencies(this._compilerHost.denormalizePath(resolvedFileName)),
     ].map((p) => p && this._compilerHost.denormalizePath(p)));
 
-    return [...uniqueDependencies]
-      .filter(x => !!x) as string[];
+    // Add used source files
+    this._usedSourceFiles.add(forwardSlashPath(fileName));
+    for (const f of esImports) {
+      this._usedSourceFiles.add(f);
```

Comment:
Is it necessary to add imports as well here? I think `getDependencies` gets called for all TS files in the compilation anyway so sources should already be added on the previous call.

Reply:
Using `getCompiledFile` should do the trick

Reply:
agreeded